simulator/virtual_machine.go

```diff
@@ -372,6 +381,36 @@ func removeDevice(devices object.VirtualDeviceList, device types.BaseVirtualDevi
 	return result
 }
 
+func (vm *VirtualMachine) genVmdkPath() (string, types.BaseMethodFault) {
+	vmdir := path.Dir(vm.Config.Files.VmPathName)
+
+	index := 0
+	for {
+		var filename string
+		if index == 0 {
+			filename = fmt.Sprintf("%s.vmdk", vm.Config.Name)
+		} else {
+			filename = fmt.Sprintf("%s_%d.vmdk", vm.Config.Name, index)
+		}
+
+		f, err := vm.createFile(vmdir, filename, false)
+		if err != nil {
+			switch err.(type) {
+			case *types.FileAlreadyExists:
+				index++
+				continue
+			default:
+				return "", err
+			}
+		}
+
+		f.Close()
+		os.Remove(f.Name())
```

Comment:
Don't want to remove the file, do you?  Otherwise, the next disk would end up as `${name}.vmdk` again, instead of `${name}_1.vmdk`

Reply:
oh, this is just to get the filename, later, `dm.createVirtualDisk` actually creates the actual disk file.

Reply:
ah right :+1: